features/stages.feature

```diff
@@ -0,0 +1,69 @@
+@stage
+Feature: Stage
+
+    Users should be allowed to tie together jobs from the same pipeline so that they can
+    convey the nature of the jobs in the workflow graph. They can either
+    set setup and teardown jobs explicitly or it will be created implicitly as a virtual job.
+
+    Eg: The following pipeline workflow:
+        (~commit) -> (hub) -> (a) -> (b) -> (c) -> (target)
+
+        Assume a, b, and c are part of a stage.
+
+    Rules:
+        - Users should be able to define pipeline stages, and add a job to a stage.
+        - A job can belong to only one stage.
+        - Relation between jobs will still be defined using requires.
+        - Jobs with empty requires (requires:[]) indicate the start of a stage.
+        - Setup and teardown jobs can be implicitly or explicitly created.
+        - When a setup/teardown job is implicitly created, it will be virtual and will not actually run.
+        - Teardown build should run when any job in a stage is terminal; downstream jobs should not continue, and stageBuild should be updated accordingly
+        - Setup build should run before a stage.
+
+    Scenario: Downstream builds are not triggered if required stage is not successful.
+        Given an existing pipeline on branch "stageFail1" with stage "simple_fail" with the workflow jobs:
+            | job       | requires      |
+            | target    | stage@simple_fail:teardown  |
+        When the "hub" job on branch "stageFail1" is started
```

Comment:
Most of the logic of this step is similar to an existing step https://github.com/screwdriver-cd/screwdriver/blob/master/features/step_definitions/trigger.js#L142C1-L167C3.
Can we use the existing step with any modifications needed?
```suggestion
        When the "hub" job on branch "stageFail1" is started
```

Reply:
updated